contracts/preset/ERC721PsiMysteryBox.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+/**
+  ______ _____   _____ ______ ___  __ _  _  _ 
+ |  ____|  __ \ / ____|____  |__ \/_ | || || |
+ | |__  | |__) | |        / /   ) || | \| |/ |
+ |  __| |  _  /| |       / /   / / | |\_   _/ 
+ | |____| | \ \| |____  / /   / /_ | |  | |   
+ |______|_|  \_\\_____|/_/   |____||_|  |_|   
+                                              
+                                            
+ */
+
+pragma solidity ^0.8.0;
+
+import "@chainlink/contracts/src/v0.8/interfaces/VRFCoordinatorV2Interface.sol";
+import "@chainlink/contracts/src/v0.8/VRFConsumerBaseV2.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+import "fpe-map/contracts/FPEMap.sol";
+
+import "../ERC721Psi.sol";
+
+abstract contract ERC721PsiMysteryBox is VRFConsumerBaseV2, ERC721Psi {
+    using Strings for uint256;
+
+    uint32 private constant _callbackGasLimit = 200000;
+
+    uint256 private seed;
+
+    event RandomnessRequest(uint256 requestId);
+
+
+    constructor() {
+        VRFConsumerBaseV2(_coordinator());
+    }
+
+    /**
+     * @dev The metadata URI before revealing.
+     */
+    function _unrevealedURI() internal pure virtual returns (string memory) {
+        return "";
+    }
+
```

Comment:
Hi @imollov  I think you can simply wrap it with a function if you want to expose it.
```
function unrevealedURI() public view  (string memory) {
    return  _unrevealedURI();
}
```
The reason I didn't expose it by default is exposing it will increase the deployment cost, and not everyone need it.

Reply:
The scenario I'm trying to achieve is the following: disable mint if revealed. The only state indicating that is the private `seed`. 

Sure I can do something like: `keccak256(abi.encodePacked(tokenURI(0))) != keccak256(abi.encodePacked(_unrevealedURI()))` but it's a workaround. 

I can also add additional state in the final contract but then I cannot be sure the VRF called back with randomness.